src/core/adapters/gitlab.js

```diff
@@ -6,14 +6,25 @@
 
 import { $has, $text } from './helpers';
 
-async function scan(loc, doc) {
-  if (doc.body.dataset.page === 'projects:issues:show') {
-    const initialDataEl = doc.getElementById('js-issuable-app-initial-data');
+function findTicketId(doc) {
+  const initialDataEl = doc.getElementById('js-issuable-app-initial-data');
```

Comment:
Purely cosmetical, but maybe it would be easier to reason about this for future developers, if we turned this around: first check for the _"new"_ structure and then have an optional fallback for the legacy approach?

This way, it could eventually also be grouped more concise into one "current" and one "legacy" block, which would make it easier to get rid of the legacy logic at a later point in time.

Maybe roughly something along the lines of this:
```js
const ticketId = doc.body.dataset.pageTypeId;

if (ticketId) {
  return ticketId;
} else {
  // Legacy approach for older self-hosted GitLab instances
  // ...
```

Reply:
The linter didn't allow me to write the if statement with an `else` clause because of the early return. That's when I started to curse at the computer and just do whatever pleases the linter. But I'll take another look at it.

Reply:
Oh, good point 💡 

I guess the linter probably wants code like this (simply skipping the `else`):
```js
const ticketId = doc...;

if (ticketId) {
  return ticketId;
}
// Legacy support
const otherId = ...
return ...
```